table/tables/cache.go

```diff
@@ -51,15 +50,14 @@ type cachedTable struct {
 	TableCommon
 	cacheData atomic.Value
 	handle    StateRemote
-	renewCh   chan func()
 	totalSize int64
 
-	mu struct {
-		sync.RWMutex
-		lockingForRead bool
-	}
+	lockingForRead tokenLimit
```

Comment:
It seems to be a semaphore, why not just use `semaphore` in go lib?

Reply:
I'm not familiar with `semaphore`, the channel is more simple for me here.